### DISCUSSION TITLE
[Windows] Virtual MIDI ports from loopMIDI do not show

### DISCUSSION THREAD

**Discussion by michael-wahler** [mido/mido#406] [Q&A]
I have successfully used `mido` with virtual MIDI ports on my Mac. Now I am trying to run my code on Windows.

To this end, I have installed `loopMidi` and created two loopback ports with it. These ports can be seen in my DAW, so I assume `loopMidi` is working correctly. However, these ports are not available to `mido`. Here is the code to reproduce the problem:

```
import mido
print (mido.get_input_names())
print (mido.get_output_names())
```

The output of this code is:

```
[]
['Microsoft GS Wavetable Synth 0']
```

Has anybody encountered this problem before?

Here is the output of `pip freeze`:

```
atomicwrites==1.4.0
attrs==21.2.0
click==8.0.0
colorama==0.4.4
dearpygui==1.8.0
distlib==0.3.6
et-xmlfile==1.1.0
filelock==3.9.0
Flask==2.0.0
formulas==1.1.0
iniconfig==1.1.1
itsdangerous==2.0.1
Jinja2==3.0.0
MarkupSafe==2.0.0
mido==1.2.10
numpy==1.20.3
numpy-financial==1.0.0
openpyxl==3.0.7
packaging==20.9
phue==1.1
platformdirs==2.6.2
pluggy==0.13.1
py==1.10.0
pynput==1.7.6
pyparsing==2.4.7
pytest==6.2.4
python-dateutil==2.8.1
python-rtmidi==1.4.9
regex==2021.4.4
schedula==1.2.6
six==1.16.0
toml==0.10.2
virtualenv==20.17.1
Werkzeug==2.0.1
```

**Comment by rdoursenaud** [comment]
Strange. I have a similar setup and tested it again to be certain. It works for me.

I assume you're running the latest version of loopMIDI: `1.0.16 (27)`

Could you please check that you are running with the `rtmidi` backend:
```py
print(mido.backend)
```
You should expect:
```py
<backend mido.backends.rtmidi (loaded)>
```

**Reply by rdoursenaud** [reply]
> Thank you for your quick response. I am using the latest version of loopMIDI.
> 
> The output of mido.backend is interesting, it says "not loaded": `<backend mido.backends.rtmidi (not loaded)>`. However, after calling `mido.get_input_names()`, it is loaded.

Yes, that is the expected behavior. Sorry I forgot to outline that.

Could you please try [another backend](https://mido.readthedocs.io/en/latest/backends/index.html?highlight=backend#choosing-a-backend) (eg. PortMIDI) just to rule out the unlikely possibility of RtMidi misbehaving?

**Comment by michael-wahler** [comment]
Thank you for your quick response. I am using the latest version of loopMIDI.

The output of mido.backend is interesting, it says "not loaded": `<backend mido.backends.rtmidi (not loaded)>`. However, after calling `mido.get_input_names()`, it is loaded.
